src/hooks/useRecentURLs.ts

```diff
@@ -0,0 +1,33 @@
+import { useLocalStorage } from 'react-use'
+
+const LOCAL_STORAGE_KEY = 'recentURLs'
+const MAX_RECENT_URLS = 3
+
+export function useRecentURLs() {
+  const [recentURLs = [], setRecentURLs] = useLocalStorage<string[]>(
+    LOCAL_STORAGE_KEY,
+    []
+  )
+
+  const addURL = (url: string) => {
+    const trimmedURL = url.trim()
+
+    if (!trimmedURL || recentURLs.includes(trimmedURL)) {
+      return
+    }
+
+    const updatedURLs = [trimmedURL, ...recentURLs].slice(0, MAX_RECENT_URLS)
```

Comment:
If you input a URL that is already at the bottom of the list, I'd expect it to be moved to the top to be considered as "most recent".

Example:

- Recent list has: `k6.io`, `quickpizza.grafana.com`, `grafana.com`.
- If you input `grafana.com` again, the list becomes `grafana.com`, `k6.io`, `quickpizza.grafana.com` (in that order).

```suggestion
    if (!trimmedURL) {
      return
    }

    const filteredURLs = recentURLs.filter((u) => u !== trimmedURL)
    const updatedURLs = [trimmedURL, ...filteredURLs].slice(0, MAX_RECENT_URLS)
```